exec/java-exec/src/main/java/org/apache/drill/exec/server/Drillbit.java

```diff
@@ -258,6 +260,13 @@ public synchronized void close() {
     }
     final Stopwatch w = Stopwatch.createStarted();
     logger.debug("Shutdown begun.");
+    // We don't really want for Drillbits to pile up in memory, so the hook should be removed
+    // It might be better to use PhantomReferences to cleanup as soon as Drillbit becomes
+    // unreachable, however current approach seems to be good enough.
+    Thread shutdownHook = this.shutdownHook;
+    if (shutdownHook != null && Thread.currentThread() != shutdownHook) {
+      Runtime.getRuntime().removeShutdownHook(shutdownHook);
```

Comment:
@vlsi `removeShutdownHook` can throw `IllegalStateException` if the virtual machine is already in the process of shutting down. Maybe we should add try / catch block to ignore such exception?

Reply:
\@arina-ielchiieva , it is handled by  `Thread.currentThread() != shutdownHook`, isn't it?

Reply:
@arina-ielchiieva , I've added try-catch

Reply:
Thanks.